core/genesis.go

```diff
@@ -246,8 +246,12 @@ func (g *Genesis) configOrDefault(ghash common.Hash) *params.ChainConfig {
 		return g.Config
 	case ghash == params.MainnetGenesisHash:
 		return params.MainnetChainConfig
-	case ghash == params.TestnetGenesisHash:
-		return params.TestnetChainConfig
+	case ghash == params.RopstenGenesisHash:
+		return params.RopstenChainConfig
+	case ghash == params.RinkebyGenesisHash:
+		return params.RinkebyChainConfig
+	case ghash == params.GoerliGenesisHash:
+		return params.GoerliChainConfig
```

Comment:
This code path is strange. Everything worked until now without having explicit clauses for Rinkeby and co. Maybe this is dead code? Perhaps we should investigate it a bit.

Reply:
I'm not entirely sure. It's used to setup genesis block in various places and detects known presets based on known genesis hashes.